meson.build

```diff
@@ -0,0 +1,362 @@
+    uint64_t x64 = 0;
+    __atomic_load_n(&x32, __ATOMIC_SEQ_CST);
+    __atomic_load_n(&x64, __ATOMIC_SEQ_CST);
+    return 0;
+    }''', name : 'built-in atomics')
+  libatomic_dep = dependency('', required: false)
+else
+  libatomic_dep = cpp.find_library('atomic')
+endif
+
+libgerbera_args = []
+
+if (cpp.has_function('nl_langinfo'))
+  libgerbera_args += '-DHAVE_NL_LANGINFO'
+endif
+
+if (cpp.has_function('setlocale'))
+  libgerbera_args += '-DHAVE_SETLOCALE'
+endif
+
+uuid_dep = dependency('uuid', required : host_machine.system() == 'linux')
+if not uuid_dep.found()
+    libgerbera_args += '-DBSD_NATIVE_UUID'
+endif
+
+# Link to libnsl (Network services library) if it exists. This is something you need on Solaris/OmniOS/Joyent
+nsl_dep = dependency('nsl', required : false)
+# Link to the socket library if it exists. This is something you need on Solaris/OmniOS/Joyent
+socket_dep = dependency('socket', required : false)
+
+if host_machine.system() == 'solaris'
+  libgerbera_args += '-DSOLARIS'
+elif host_machine.system() != 'linux'
+  libgerbera_args += '-DBSD'
+endif
+
+if get_option('debug')
+    libgerbera_args += '-DTOMBDEBUG'
+    libgerbera_args += '-DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_DEBUG'
+else
+    libgerbera_args += '-DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_INFO'
+endif
+
+avcodec_dep = dependency('libavcodec', required : get_option('avcodec'))
+avfilter_dep = dependency('libavfilter', required : get_option('avcodec'))
+avformat_dep = dependency('libavformat', required : get_option('avcodec'))
+avutil_dep = dependency('libavutil', required : get_option('avcodec'))
+if avcodec_dep.found() and avfilter_dep.found() and avformat_dep.found() and avutil_dep.found()
+  libgerbera_args += '-DHAVE_FFMPEG'
+  if cpp.has_member('struct AVStream', 'codecpar', prefix : '#include <libavformat/avformat.h>', dependencies : avformat_dep)
+    libgerbera_args += '-DHAVE_AVSTREAM_CODECPAR'
+  endif
+endif
+
+curl_dep = dependency('libcurl', required : get_option('curl'))
+if curl_dep.found()
+  libgerbera_args += '-DHAVE_CURL'
+  libgerbera_args += '-DONLINE_SERVICES'
+  libgerbera_args += '-DATRAILERS'
+endif
+
+exif_dep = dependency('libexif', required : get_option('exif'))
+if exif_dep.found()
+  libgerbera_args += '-DHAVE_LIBEXIF'
+endif
+
+exiv2_dep = dependency('exiv2', required : get_option('exiv2'))
+if exiv2_dep.found()
+  libgerbera_args += '-DHAVE_EXIV2'
+  if exiv2_dep.version().version_compare('<0.28')
+    libgerbera_args += '-D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR'
+  endif
+endif
+
+ffmpegthumbnailer_dep = dependency('libffmpegthumbnailer', required : get_option('ffmpegthumbnailer'))
+if ffmpegthumbnailer_dep.found()
+  libgerbera_args += '-DHAVE_FFMPEGTHUMBNAILER'
+endif
+
+inotify_dep = dependency('', required : false)
+if not get_option('inotify').disabled()
+  if host_machine.system() != 'linux' and host_machine.system() != 'haiku'
+    inotify_dep = dependency('inotify')
+  endif
+  libgerbera_args += '-DHAVE_INOTIFY'
+endif
```

Comment:
When using feature options, you want the auto case to not be exactly equal to the enabled case. That means passing `required: get_option('inotify')` to the non-linux dependency lookup.

Since on Linux it won't be a dependency and is just "there", you will want to set a boolean inotify_found, e.g.

```

if host_machine.system() == 'linux' or host_machine.system() == 'haiku'
    inotify_dep = dependency('', required : false)
    inotify_found = not get_option('inotify').disabled()
else
    inotify_dep = dependency('inotify', required: get_option('inotify'))
    inotify_found = inotify_dep.found()
endif

if inotify_found
    libgerbera_args += '-DHAVE_INOTIFY'
endif
```